### PR TITLE
Presto-Redis connector

### PR DESCRIPTION
**Pull request by avasilevskiy**
The connector allow querying of key/value Redis data where value is of the String type. See doc file for more details.

There's code duplication between Redis and Kafka connector in the decoders. This will be commoned in the subsequent pull request as I prefer not to check in new functionality and code refactoring in single shot.

### CODE REVIEW

**Comment by snarayananqubole** [comment]
There is another open PR for a kinesis connector that also duplicates decoder stuff. Might be good to coordinate https://github.com/facebook/presto/pull/2885

**Comment by avasilevskiy** [comment]
Ok, great. I'd recommend to merge these requests and create a separate pull request for code refactoring

**Comment by hgschmie** [comment]
I had a couple of conversations to pull out the decoders and the static metadata configuration into presto proper (as I wrote the initial version of that).

**Comment by cberner** [comment]
I think we should do the refactoring first, as that will make the code easier to review. Could you split this into two PRs, one that refactors the Kafka connector, and then one that adds Redis connector?

**Comment by avasilevskiy** [comment]
cberner, I've added the refactored code. There are many different possibilities where the commoned-up code could live (presto-main, presto-spi, presto-util,....) each with different pros and cons. I've decided to put the common part in the presto-spi since at the moment it's the place where most of the other connector-common parts reside. One of the downsides is that presto-spi now has more dependencies than before.

Please take a look

**Comment by electrum** [comment]
We should add a new module for this as an optional utility library for connectors.

**Comment by avasilevskiy** [comment]
How does presto-util sound ?

File: presto-redis/src/main/java/com/facebook/presto/redis/RedisSplitManager.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                             RedisJedisManager jedisManager)
+    {
+        this.connectorId = checkNotNull(connectorId, "connectorId is null");
+        this.redisConnectorConfig = checkNotNull(redisConnectorConfig, "redisConfig is null");
+        this.handleResolver = checkNotNull(handleResolver, "handleResolver is null");
+        this.jedisManager = checkNotNull(jedisManager, "jedisManager is null");
+    }
+
+    @Override
+    public ConnectorPartitionResult getPartitions(ConnectorTableHandle tableHandle, TupleDomain<ColumnHandle> tupleDomain)
```
Comment (martint) [inline presto-redis/src/main/java/com/facebook/presto/redis/RedisSplitManager.java]: The `ConnectorPartition` abstraction is deprecated (it's been replaced by TableLayout -- see TpchMetadata for an example of how it's used)

**Comment by cberner** [comment]
I'd name it "presto-connector-utils", but will defer to @electrum on the naming

**Comment by avasilevskiy** [comment]
@martint, I've pulled decoders into a separate module and removed deprecated classes. Can you look into integrating this PR ? thx

**Comment by electrum** [comment]
Sorry for the delay on this. We've been really busy and will try to get to it soon.

**Comment by dain** [comment]
Merged, thanks!

**Pull request closed without merging**